src/client/Microsoft.Identity.Client.Brokers/WamAdapters.cs

```diff
@@ -0,0 +1,236 @@
+            {
+                authParams.DecodedClaims = authenticationRequestParameters.ClaimsAndClientCapabilities;
+            }
+
+            //pass extra query parameters if there are any
+            if (authenticationRequestParameters.ExtraQueryParameters != null)
+            {
+                foreach (KeyValuePair<string, string> kvp in authenticationRequestParameters.ExtraQueryParameters)
+                {
+                    authParams.Properties[kvp.Key] = kvp.Value;
+                }
+            }
+
+            return authParams;
+        }
+
+        /// <summary>
+        /// Parse Native Interop AuthResult Response to MSAL Token Response
+        /// </summary>
+        /// <param name="authResult"></param>
+        /// <param name="authenticationRequestParameters"></param>
+        /// <param name="logger"></param>
+        /// <returns></returns>
+        /// <exception cref="MsalServiceException"></exception>
+        public static MsalTokenResponse ParseRuntimeResponse(
+                NativeInterop.AuthResult authResult, 
+                AuthenticationRequestParameters authenticationRequestParameters,
+                ICoreLogger logger)
+        {
+            try
+            {
+                string expiresOn = authResult.ExpiresOn.ToString();
+                string correlationId = authenticationRequestParameters.CorrelationId.ToString("D");
+
+                if (string.IsNullOrWhiteSpace(correlationId))
+                {
+                    logger.Warning("No correlation ID in response");
+                    correlationId = null;
+                }
+
+                MsalTokenResponse msalTokenResponse = new MsalTokenResponse()
+                {
+                    AccessToken = authResult.AccessToken,
+                    IdToken = authResult.RawIdToken,
+                    CorrelationId = correlationId,
+                    Scope = authResult.GrantedScopes,
+                    ExpiresIn = DateTimeHelpers.GetDurationFromWindowsTimestamp(expiresOn, logger),
+                    ClientInfo = authResult.Account.ClientInfo.ToString(),
+                    TokenType = "Bearer",
+                    WamAccountId = authResult.Account.Id,
+                    TokenSource = TokenSource.Broker
+                };
+
+                logger.Info("WAM response status success");
+
+                return msalTokenResponse;
+            }
+            catch (NativeInterop.MsalRuntimeException ex)
+            {
+                throw new MsalServiceException("wam_failed", $"Could not acquire token using WAM. {ex.Message}");
+            }
+
+        }
+
+        /// <summary>
+        /// Get WAM Application Redirect URI
+        /// </summary>
+        /// <param name="clientId"></param>
+        /// <returns></returns>
+        private static string GetExpectedRedirectUri(string clientId)
+        {
+#if WINDOWS_APP
```

Comment:
We don't compile this dll on UWP, so this #if is dead code.

Reply:
removed dead code